### PR TITLE
ref: Clean up client error handling across plugins

### PR DESCRIPTION
**Pull request by dcramer**
- Move GitHub to api client abstractions
- Move Bitbucket to api client abstractions
- Clean up various duplicated code
- Add basic unit tests for ApiClient and AuthApiClient

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dcramer** [comment]
Will get tests passing and review coverage to make sure I feel confident, but if anyone has feedback on overall direction lmk

File: src/sentry_plugins/asana/plugin.py [hunk 1]
```diff
@@ -155,7 +130,7 @@ def create_issue(self, request, group, form_data, **kwargs):
                 workspace=self.get_option('workspace', group.project), data=form_data
             )
         except Exception as e:
-            self.raise_error(e)
+            self.raise_error(e, identity=client.auth)
```
Comment (dcramer) [inline src/sentry_plugins/asana/plugin.py]: this is the main goal here, so we can attach the identity to 'InvalidIdentity' errors

Comment (MaxBittker) [reply src/sentry_plugins/asana/plugin.py]: thanks, helpful

**Comment by dcramer** [comment]
looks like master is broken, probably related to some change in sentry, looking into it now

**Comment by dcramer** [comment]
Shipping this as its breaking build

**Comment by MaxBittker** [comment]
i reviewed half of it, lgtm

**Comment by dcramer** [comment]
im opening another PR as I wanna review it myself before the huge swath of changes end up in master (didnt mean to merge htis one)

**Comment by dcramer** [comment]
https://github.com/getsentry/sentry-plugins/pull/257

**Pull request merged**